sql/plan/alter_table.go

```diff
@@ -722,6 +723,28 @@ func (i *dropColumnIter) Next(ctx *sql.Context) (sql.Row, error) {
 		return nil, err
 	}
 
+	// drop constraints that only reference the dropped column
+	cat, ok := i.alterable.(sql.CheckAlterableTable)
+	if !ok {
+		return nil, errors.New("failed to cast to CheckAlterableTable")
+	}
+
+	for _, check := range i.d.Checks {
```

Comment:
(minor) might be worth a comment here just to let future code-readers know that validation was done earlier, so we know that any check constraints that still reference this column are safe to delete.